src/workers/postUpdated.ts

```diff
@@ -75,76 +292,49 @@ const worker: Worker = {
+        let combinedPost: Partial<ArticlePost>;
+        if (!post_id) {
+          // Handle creation of new post
+          try {
+            combinedPost = await createPost({
+              logger,
+              entityManager,
+              data: fixedData,
+              submission_id: data?.submission_id,
+            });
+          } catch (e) {
+            return;
+          }
+        } else {
+          // Handle update of existing post
+          combinedPost = await updatePost({
+            entityManager,
+            data: fixedData,
+            id: post_id,
+          });
+        }
+
+        // After add or update:
+        if (combinedPost.visible) {
+          // Update all referring posts to become visible
+          await entityManager.getRepository(SharePost).update(
+            { sharedPostId: combinedPost.id },
             {
-              url: data.url,
-              keywords: allowedKeywords,
+              visible: true,
+              visibleAt: combinedPost.visibleAt,
+              private: combinedPost.private,
             },
-            'created an article with more than 5 keywords',
           );
         }
 
-        const { private: privacy } = await entityManager
-          .getRepository(Source)
-          .findOneBy({ id: data?.source_id });
-
-        const fixedData: Partial<ArticlePost> = {
-          origin: data?.origin as PostOrigin,
-          authorId,
-          creatorTwitter,
-          url: data?.url,
-          canonicalUrl: data?.extra?.canonical_url || data?.url,
-          image: data?.image,
-          sourceId: data?.source_id,
-          title: title && he.decode(title),
-          readTime: parseReadTime(data?.extra?.read_time),
-          publishedAt: data?.published_at && new Date(data?.published_at),
-          metadataChangedAt: updatedDate,
-          visible: becomesVisible,
-          visibleAt: becomesVisible
-            ? databasePost.visibleAt ?? updatedDate
-            : null,
-          tagsStr: allowedKeywords?.join(',') || null,
-          private: privacy,
-          sentAnalyticsReport: privacy || !authorId,
-          summary: data?.extra?.summary,
-          description: data?.extra?.description,
-          siteTwitter: data?.extra?.site_twitter,
-          toc: data?.extra?.toc,
-          contentCuration: data?.extra?.content_curation,
-        };
-
-        await entityManager
-          .getRepository(ArticlePost)
-          .update({ id: databasePost.id }, fixedData);
-
-        if (becomesVisible) {
-          // Update all reffering posts to become visible
-          await entityManager
-            .getRepository(SharePost)
-            .update(
-              { sharedPostId: databasePost.id },
-              { visible: true, visibleAt: updatedDate, private: privacy },
-            );
-        }
-
-        await addKeywords(entityManager, mergedKeywords, data.post_id);
+        await addKeywords(entityManager, mergedKeywords, combinedPost.id);
```

Comment:
This should also happen on creation and if the tags were changed since last time.
Also bare in mind that we probably have to delete the old keywords and apply the new ones

Reply:
This happens for both flows, I didn't change the internals of this.